WordPress/src/test/java/org/wordpress/android/ui/posts/editor/EditorActionsProviderTest.kt

```diff
@@ -0,0 +1,172 @@
+package org.wordpress.android.ui.posts.editor
+
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.times
+import com.nhaarman.mockitokotlin2.verify
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.ArgumentMatchers.any
+import org.mockito.junit.MockitoJUnitRunner
+import org.wordpress.android.fluxc.model.post.PostStatus
+import org.wordpress.android.util.CrashLoggingUtilsWrapper
+
+@RunWith(MockitoJUnitRunner::class)
+class EditorActionsProviderTest {
+    private var crashLoggingUtilsWrapper: CrashLoggingUtilsWrapper = mock()
+    private val actionsProvider: EditorActionsProvider = EditorActionsProvider(
+            crashLoggingUtilsWrapper
+    )
+
+    @Test
+    fun `All secondary actions but NONE have isVisible set to TRUE`() {
+        for (action in SecondaryEditorAction.values()) {
+            assertThat(action.isVisible).isEqualTo(action != SecondaryEditorAction.NONE)
+        }
+    }
+
+    @Test
+    fun `Verify actions of a DRAFT`() {
+        // Arrange & Act
+        val (primaryAction, secondaryAction) =
+                getPrimaryAndSecondaryActions(PostStatus.DRAFT, userCanPublish = true)
+        // Assert
+        assertThat(primaryAction).isEqualTo(PrimaryEditorAction.PUBLISH_NOW)
+        assertThat(secondaryAction).isEqualTo(SecondaryEditorAction.SAVE)
+    }
+
+    @Test
+    fun `Verify actions of a DRAFT when the user doesn't have publishing rights`() {
+        // Arrange & Act
+        val (primaryAction, secondaryAction) =
+                getPrimaryAndSecondaryActions(PostStatus.DRAFT, userCanPublish = false)
+        // Assert
+        assertThat(primaryAction).isEqualTo(PrimaryEditorAction.SUBMIT_FOR_REVIEW)
+        assertThat(secondaryAction).isEqualTo(SecondaryEditorAction.NONE)
+    }
+
+    @Test
+    fun `Verify actions of a PUBLISHED post`() {
+        // Arrange & Act
+        val (primaryAction, secondaryAction) =
+                getPrimaryAndSecondaryActions(PostStatus.PUBLISHED, userCanPublish = true)
+        // Assert
+        assertThat(primaryAction).isEqualTo(PrimaryEditorAction.UPDATE)
+        assertThat(secondaryAction).isEqualTo(SecondaryEditorAction.NONE)
+    }
+
+    @Test
+    fun `Verify an error is logged when the user doesn't have publishing rights and works with a PUBLISHED post`() {
+        // Arrange & Act
+        val (_, _) =
```

Comment:
Hi, @malinajirka do we have to create a variable here to store the result if we aren't using it?  The names have also been omitted for brevity, so I want to understand what's the intent of creating it if it's to communicate that a result is being returned. Let me know. Thanks 😄

Reply:
Good point! ;D Removed in 19492ed4c6166f9254f03c4b6a2d79691ca70ff3